### PR TITLE
Correction lesson controller

### PR DESCRIPTION
**Pull request by Dmitryshl**

### CODE REVIEW

File: spec/swagger_helper.rb [hunk 1]
```diff
@@ -103,9 +105,16 @@
...
+            category_id: { type: :integer, minimum: 1 }
           },
           required: %w[title description status video_link author_id]
+        },
+        delete_lesson: {
+          type: :object,
+          properties: {
+            id: { type: :integer, minimum: 1 }
+          }
         }
```
Comment (Terapool) [inline spec/swagger_helper.rb:118]: required: %w[id]

File: spec/swagger_helper.rb [hunk 2]
```diff
@@ -79,7 +79,8 @@
             description: { type: :string },
             video_link: { type: :string },
             status: { type: :string },
-            author_id: { type: :integer, minimum: 1 }
+            author_id: { type: :integer, minimum: 1 },
+            category_id: { type: :integer, minimum: 1 }
           },
           required: %w[title description status video_link author_id]
```
Comment (Terapool) [inline spec/swagger_helper.rb:85]: category_id is required here or not?

File: app/views/lessons/show.jbuilder [hunk 3]
```diff
@@ -3,4 +3,6 @@ json.post do
   json.title @lesson.title
   json.description @lesson.description
   json.video_link @lesson.video_link
+  json.author_id @lesson.author_id
```
Comment (Terapool) [inline app/views/lessons/show.jbuilder:6]: You need to remove 'post' from here. Now you return something like this:

{
  "post": {
    "id": 6,
    "title": "string",
    "description": "string",
    "video_link": "string",
    "author_id": 1,
    "category_id": 1
  }
}

See app/views/categories/show.jbuilder as example

File: spec/requests/lessons_spec.rb [hunk 4]
```diff
@@ -54,7 +55,8 @@
           description: 'Introduce',
           vvideo_link: 'link',
```
Comment (Terapool) [inline spec/requests/lessons_spec.rb:56]: typo: 'vvideo_link'

File: spec/requests/lessons_spec.rb [hunk 5]
```diff
@@ -54,7 +55,8 @@
...
           status: 'status',
```
Comment (Terapool) [inline spec/requests/lessons_spec.rb:57]: Actually status is not returned after creation...

File: spec/requests/lessons_spec.rb [hunk 6]
```diff
@@ -76,7 +78,8 @@
...
           author_id: 1,
-          id: 1
+          id: 1,
```
Comment (Terapool) [inline spec/requests/lessons_spec.rb:81]: You can not update id in lesson!!!

**Review by Terapool — CHANGES_REQUESTED** [review]
Too many bugs in Lessons requests and corresponding schema

File: app/views/lessons/show.jbuilder [hunk 7]
```diff
@@ -1,6 +1,7 @@
-json.post do
-  json.id @lesson.id
-  json.title @lesson.title
-  json.description @lesson.description
-  json.video_link @lesson.video_link
-end
+json.id @lesson.id
```
Comment (Terapool) [inline app/views/lessons/show.jbuilder:1]: Why not use json.extract! here?

File: app/controllers/lessons_controller.rb [hunk 8]
```diff
@@ -13,9 +13,9 @@ def new
   end
 
   def create
-    @lesson = current_user.lessons.new(lesson_params)
+    @lesson = Lesson.new(lesson_params)
     if @lesson.save
-      redirect_to @lesson
+      render 'lessons/show'
```
Comment (evrusha) [inline app/controllers/lessons_controller.rb:18]: Just `render :show` ?

File: app/views/lessons/show.jbuilder [hunk 9]
```diff
@@ -1,6 +1,7 @@
...
-  json.description @lesson.description
-  json.video_link @lesson.video_link
-end
+json.extract! @lesson, :id
+json.title @lesson.title
+json.description @lesson.description
+json.video_link @lesson.video_link
+json.author_id @lesson.author_id
+json.category_id @lesson.category_id
+json.status @lesson.status
```
Comment (evrusha) [inline app/views/lessons/show.jbuilder:7]: Better this? `json.extract! @lesson, :id, :title, :description, :video_link, :author_id, :category_id, :status`

File: app/controllers/lessons_controller.rb [hunk 10]
```diff
@@ -32,16 +32,20 @@ def update
...
   private
 
   def lesson_params
-    params.require(:lesson).permit(:title, :description, :status, :video_link, :author_id, :category_id)
+    params.permit(:title, :description, :status, :video_link, :author_id, :category_id)
   end
 
   def lesson_find
-    @lesson = Lesson.all.find(params[:id])
+    @lesson = Lesson.all.find_by(id: params[:id])
```
Comment (evrusha) [inline app/controllers/lessons_controller.rb:49]: Why we need `all`?

File: app/controllers/lessons_controller.rb [hunk 11]
```diff
@@ -13,9 +13,9 @@ def new
   end
 
   def create
-    @lesson = current_user.lessons.new(lesson_params)
+    @lesson = Lesson.new(lesson_params)
     if @lesson.save
-      redirect_to @lesson
+      render :show
     else
       render :new, status: :unprocessable_entity
```
Comment (fooooxmr) [inline app/controllers/lessons_controller.rb:20]: We do not need to render show here. We need to return an errors with validations or anything else.

File: app/views/lessons/error.jbuilder [hunk 12]
```diff
@@ -0,0 +1 @@
+json.deleted false
```
Comment (fooooxmr) [inline app/views/lessons/error.jbuilder:1]: Errors should be more meaningful and should be reused for all errors. Add error message here.

**Review by fooooxmr — CHANGES_REQUESTED** [review]

**Review by Terapool — APPROVED** [review]

**Review by olegbrest1 — APPROVED** [review]

**Review by fooooxmr — APPROVED** [review]

**Review by evrusha — APPROVED** [review]

**Pull request merged**